nservicebus/sagas/saga-not-found.md

```diff
@@ -29,3 +29,74 @@ The ability to provide an implementation for `IHandleSagaNotFound` is especially
 For example, consider a saga that is used for managing the registration process on the website. After a customer registers, they receive an email with a confirmation link. The system will wait for confirmation for a specific period of time, e.g. 24 hours. If the user doesn't click the link within 24 hours, their data is removed from the system and saga is completed. However, they might decide to click the confirmation link a few days later. In this case, the related saga instance can't be found and an exception will be thrown. By implementing `IHandleSagaNotFound` it is possible to handle the situation differently, e.g. redirect the user to the registration website and ask them to fill out the form again.
 
 The implementation of `IHandleSagaNotFound` should be driven by the business requirements for a specific situation. In some cases the message might be ignored; in others, it might be useful to track whenever that situation happens (e.g. by logging or sending another message). In still other cases, it might make sense to perform a custom compensating action.
+
+## Troubleshooting
+
+*Saga not found* Exceptions can occur when:
+
+1. The saga instance does not exist (yet)
+2. The saga instance has already been removed
+
+It's not always obvious when saga state does or does not exist. Most often the cause is due to race conditions.
+
+Note: Expect out-of-order delivery if the message flow allows it and expect more-than-once processing.
+
+### Out of order
+
+If the design assumes message A will create the saga and message B updates the saga. If message B is received and processed before message A, the saga doesn't exist yet, and the message will be discarded.
+
+Symptoms:
+
+- Sagas are expected to exist but this does not always seems to be the case. Inspecting storage shows that the saga does exist.
+
+Mitigation:
+
+- Use [IAmStartedBy<>](/nservicebus/sagas/#starting-a-saga) instead of `IHandleMessages<>` for **any** message type which can originate from outside of the saga. No matter in which order messages are delivered, any message type processed first must create the saga.
+
```

Comment:
```suggestion
- Use [IAmStartedBy<>](/nservicebus/sagas/#starting-a-saga) instead of `IHandleMessages<>` for **any** message type which can originate from outside of the saga. No matter in which order messages are delivered, any message type processed first must create the saga instance.
```